examples/hello_coro.cpp

```diff
@@ -13,7 +13,7 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-#if defined(__GNUC__) && !defined(__clang__)
```

Comment:
If we have determined that coroutines are unusable on gcc-11 and earlier, then we should probably just set `_STD_NO_COROUTINES` to 1 in `stdexec/coroutine.hpp` and remove this check here. Line 25 below checks `_STD_NO_COROUTINES`.

Reply:
Will do.

Reply:
Actually, I don't think it is that co-routines are unusable on gcc-11 and earlier.  With gcc-11, we get the following compile failure with something in task.hpp:

```
In file included from /github/workspace/examples/hello_coro.cpp:26:
/github/workspace/include/exec/task.hpp:121:43: error: redefinition of ‘struct exec::__task::__default_task_context_impl::__awaiter_context<_ParentPromise>’
  121 |       struct __default_task_context_impl::__awaiter_context<_ParentPromise> {
      |                                           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/github/workspace/include/exec/task.hpp:93:43: note: previous definition of ‘struct exec::__task::__default_task_context_impl::__awaiter_context<_ParentPromise>’
   93 |       struct __default_task_context_impl::__awaiter_context<_ParentPromise> {
      |                                           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/github/workspace/include/exec/task.hpp:134:43: error: redefinition of ‘struct exec::__task::__default_task_context_impl::__awaiter_context<_ParentPromise>’
  134 |       struct __default_task_context_impl::__awaiter_context<_ParentPromise> {
      |                                           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/github/workspace/include/exec/task.hpp:93:43: note: previous definition of ‘struct exec::__task::__default_task_context_impl::__awaiter_context<_ParentPromise>’
   93 |       struct __default_task_context_impl::__awaiter_context<_ParentPromise> {
      |                                           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

I don't think setting `_STD_NO_COROUTINES` on gcc-11 is entirely appropriate.  Any ideas on how we might resolve this issue?  It isn't obvious to me what gcc is unhappy about.

Reply:
You're right. This is a gcc bug with concepts. I can work around it, and I've confirmed locally that gcc-11 can handle coroutines just fine. I'll make a PR and merge, then this PR is not needed.